templates/publish.yml

```diff
@@ -0,0 +1,118 @@
+    registry1Password:
+      default: ${IRON_BANK_ROBOT_PASSWORD}
+    chainguardIdentity:
+      default: ${CHAINGUARD_IDENTITY}
+    targetRepo:
+      default: ""
+    team:
+      default: uds
+---
+
+publish:
+  variables:
+    GIT_STRATEGY: clone
+    GIT_DEPTH: 0
+  id_tokens:
+    CHAINGUARD_TOKEN:
+      aud: https://console-api.enforce.dev
+  rules:
+    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
+    - if: $CI_MERGE_REQUEST_ID
+      variables:         # Override DRY_RUN defined
+        DRY_RUN: "true"  # at the job level.
+  script:
+    # Dry run on non-tag runs
+    - OPTIONS="--no-progress"
+    - if [[ "$DRY_RUN" == "true" ]]; then OPTIONS="$OPTIONS --dry-run"; fi
+
+    # Environment setup
+    - |
+      uds run actions:authenticate-registries --set REGISTRY1_USERNAME="$[[ inputs.registry1Username ]]" \
+                                              --set REGISTRY1_PASSWORD="$[[ inputs.registry1Password ]]" \
+                                              --set CHAINGUARD_IDENTITY="$[[ inputs.chainguardIdentity ]]" \
+                                              --set CHAINGUARD_TOKEN="$CHAINGUARD_TOKEN" \
+                                              --set GITLAB_REGISTRY_URL="$CI_REGISTRY" \
+                                              --set GITLAB_REGISTRY_USER="$CI_REGISTRY_USER" \
+                                              --set GITLAB_REGISTRY_TOKEN="$CI_REGISTRY_PASSWORD"
+
+    # Check if release is necessary for flavor and exit early if not
+    - |
+      if [[ "$DRY_RUN" == "true" ]]; then
+        echo "Dry run, skipping check if release is required: uds-releaser check ${FLAVOR}"
+      else
+        if ! uds-releaser check ${FLAVOR}; then echo "No release necessary"; exit 0; fi
+      fi
+
+    # Modify zarf package and uds bundle to have the correct version
+    - |
+      if [[ "$DRY_RUN" == "true" ]]; then
+        echo "Dry run, skipping release command: uds-releaser update-yaml ${FLAVOR}"
+      else
+        uds-releaser update-yaml ${FLAVOR}
+      fi
+
+    # Publish Package
+    - |
+        if uds run --list | grep -q 'publish-package'; then
+          TASK_NAME="publish-package"
+        else
+          TASK_NAME="publish-release"
+        fi
+
+        if [[ -n "$[[ inputs.targetRepo ]]" ]]; then
+          uds run $TASK_NAME --set FLAVOR="$[[ inputs.flavor ]]"\
+                                     --set VERSION="$(uds zarf tools yq ".flavors[] | select(.name == \"${FLAVOR}\") | .version" releaser.yaml)"\
+                                     --set TARGET_REPO="$[[ inputs.targetRepo ]]"\
+                                     ${OPTIONS}
+        else
+          uds run $TASK_NAME --set FLAVOR="$[[ inputs.flavor ]]"\
+                                     --set VERSION="$(uds zarf tools yq ".flavors[] | select(.name == \"${FLAVOR}\") | .version" releaser.yaml)"\
+                                     --set TEAM="$[[ inputs.team ]]"\
+                                     ${OPTIONS}
+        fi
```

Comment:
rather than copying all of this we might be able to simplify into something like how dry-run and options interact

Reply:
I will do this but want to resolve the `TARGET_REPO` conversation above first because I might just delete that.

Reply:
Fixed it up!